cmd/network-passt-binding/server/server.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package server
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	vmschema "kubevirt.io/api/core/v1"
+
+	hooksInfo "kubevirt.io/kubevirt/pkg/hooks/info"
+	hooksV1alpha2 "kubevirt.io/kubevirt/pkg/hooks/v1alpha2"
+
+	"kubevirt.io/client-go/log"
+
+	"kubevirt.io/kubevirt/pkg/network/istio"
+
+	"kubevirt.io/kubevirt/cmd/network-passt-binding/callback"
+	"kubevirt.io/kubevirt/cmd/network-passt-binding/domain"
+)
+
+type InfoServer struct {
+	Version string
+}
+
+func (s InfoServer) Info(_ context.Context, _ *hooksInfo.InfoParams) (*hooksInfo.InfoResult, error) {
+	log.Log.Infof("Info method has been called")
```

Comment:
Well, I see it as an extreme.
We should care if something has changed, not if something was called. It is bad practice to use logs to perform call traces. The same reasoning given here one can use to just record any call to a function in general.

Reply:
Right, this is why we dont record any function only the one that interesting and can help debug / give indication that it works.
This is no different then any other server log that records requests.

Reply:
The info callback log message enable verifying the 
The most impotent one is the OnDefineDomain message, it enable verifying the request actually reached the sidecar container.
Same for Info callback, it enable verifying the grpc server started inside the sidecar container started.

These are systematic messages that give indication the hook server actually works works on the sidecar container.
I think it has value because it could help debugging in integration and in production.
It has with almost zero footprint and w/o spamming virt-launcher logs, those messages will be seen in the sidecar container logs.

Reply:
> It has with almost zero footprint and w/o spamming virt-launcher logs, those messages will be seen in the sidecar container logs.

These messages makes the code look as if we need to add function traces to check the application operation.
We need to log only action that has impact on the functionality, e.g. that a mutation has been performed on the domain configuration or warnings in case something is not as expected.

It is also fine to log the initial start of the sidecar with any information relevant, or its termination.
Filling up logs with logs about functions being called is not something we are suppose to have in production code.

Please try to convince me here with a real scenario where these logs will help troubleshooting the problem.
if the sidecar is still running or not, can be seen from the pod status (e.g. if it crashed). If grpc is somehow not working, we are suppose to get errors from the grpc infra (on the virt-launcher side).
I am even fine to report if the domain changed, with the original and target config, or if it did not changed because 1,2,3...

Reply:
Thanks for the discussion, here and offline, I removed the redundent logs messages.